app/controllers/grades_controller.rb

```diff
@@ -478,9 +466,15 @@ def update_Files
       f.json { autosave_comments(comments_params, :comment_to_inlinecomment) }
       f.html {
         save_all_comments(comments_params, :comment_to_inlinecomment)
-        mark_grading_allocation_completed
-        redirect_to course_assignment_submission_path(@course, @assignment, @submission),
-                    notice: "Comments saved; grading completed"
+        comments = InlineComment.where(submission_id: @submission.id)
+        if comments.any? {|a| a.comment.empty? } 
+          return redirect_back fallback_location: course_assignment_path(@course, @assignment),
```

Comment:
This is exactly the scenario we have to be careful to avoid: merely redirecting back to the prior page will abandon *all* the comments, including the perfectly-fine ones.